docs/changelog/0.4.0.md

```diff
@@ -0,0 +1,531 @@
+```scala
+nativeConfig := nativeConfig.value.withLTO(build.LTO.thin)
+
+nativeLTO := "none"
+```
+
+### CFuncPtr syntax change
+You no longer need to implement the `CFuncPtrN` trait which is now private for Scala Native implementation.
+Instead, you can use an implicit conversion method taking arbitrary `scala.FunctionN` and returning `CFuncPtrN`.
+
+```scala
+type Callback = CFuncPtr1[CInt,Unit]
+def registerCallback(cFn: Callback) = extern
+def fn(n: CInt): Unit = ???
+
+registerCallback(CFuncPtr1.fromScalaFunction(fn))
+registerCallback(fn)
+
+registerCallback{n: CInt => println("hello native")}
+```
+
+It's now also possible to work with an arbitrary pointer and convert it to `CFuncPtrN` that can be called in your Scala code
+or to convert your function to any pointer if your native library needs this.
+
+```scala
+import scala.scalanative.unsafe.Ptr
+val cFnPtr: CFuncPtr0[CInt] = ???
+val fnPtr: Ptr[Byte]    = CFuncPtr.toPtr(cFnPtr)
+val fnFromPtr:       = CFuncPtr.fromPtr[CFuncPtr0[CInt]](fnPtr)
+```
+
+### Other breaking changes:
+* Sbt 0.13.x is no longer supported - upgrade to 1.1.6 or newer.
+* The minimal version of Clang working with Scala Native is now 6.0
+* `CSize` is now unsigned numeral type
+* Usage of signed numeral types for methods expecting `CSize` was deprecated. 
+
+## New features
+
+### Supported Scala versions
+We added support for Scala 2.12.13 and 2.13.4, in addition to the existing support for 2.11.12. 
+
+### JUnit Native Plugin
+Scala Native now comes with JUnit support out of the box, this means that you can write tests in the same way you would do
+for a Scala/JVM or Scala.js project. To enable JUnit tests all you will need to do is to add the two following lines to your `build.sbt`.
+
+```scala
+addCompilerPlugin("org.scala-native" % "junit-plugin" % SNVersion cross CrossVersion.full)
+libraryDependencies += "org.scala-native" % "junit-runtime" %%% SNVersion % "test"
+```
+
+### Reflective instantiation
+Since this release you are able to reflectively instantiate definitions marked with the `@EnableReflectiveInstantation` annotation.
+It will also make all of its descendants able to be reflectively instantiated.
+For definitions marked with this annotation Scala Native plugin would generate definitions of static class initializers
+and invoke them just before the `main` method in your final program.
+This step would register `LoadableModuleClass` and `InstantiableClass` instances for objects and classes accordingly allowing
+them to be accessed later via the provided `scalanative.reflect.Reflect` object methods in your code.
+If you have used Scala.js before, it may seem similar to you, as the new implementation uses exactly the same API.
+
+Scala Native does not support full reflection support, although this feature might fix most of the issues that could occur
+in users code.
+
+```scala
+package x.y.z
+
+@EnableReflectiveInstantation
+sealed trait ReflectiveFoo {
```

Comment:
Consider removing `sealed`, otherwise it suggests that this *only* works for sealed traits:
```suggestion
@EnableReflectiveInstantation
trait ReflectiveFoo {
```